pkg/identity/revocation_db.go

```diff
@@ -22,25 +22,73 @@ type RevocationDB struct {
 	DB storage.KeyValueStore
 }
 
+// NewRevocationDB returns a new revocation database given the URL
+func NewRevocationDB(revocationDBURL string) (*RevocationDB, error) {
+	driver, source, err := dbutil.SplitConnstr(revocationDBURL)
+	if err != nil {
+		return nil, extensions.ErrRevocationDB.Wrap(err)
+	}
+
+	var db *RevocationDB
+	switch driver {
+	case "bolt":
+		db, err = newRevocationDBBolt(source)
+		if err != nil {
+			return nil, extensions.ErrRevocationDB.Wrap(err)
+		}
+	case "redis":
+		db, err = newRevocationDBRedis(revocationDBURL)
+		if err != nil {
+			return nil, extensions.ErrRevocationDB.Wrap(err)
+		}
+	default:
+		return nil, extensions.ErrRevocationDB.New("database scheme not supported: %s", driver)
+	}
+
+	return db, nil
+}
+
+// newRevocationDBBolt creates a bolt-backed RevocationDB
+func newRevocationDBBolt(path string) (*RevocationDB, error) {
+	client, err := boltdb.New(path, extensions.RevocationBucket)
+	if err != nil {
+		return nil, err
+	}
+	return &RevocationDB{
+		DB: client,
+	}, nil
+}
+
+// newRevocationDBRedis creates a redis-backed RevocationDB.
+func newRevocationDBRedis(address string) (*RevocationDB, error) {
+	client, err := redis.NewClientFrom(address)
+	if err != nil {
+		return nil, err
+	}
+	return &RevocationDB{
+		DB: client,
+	}, nil
+}
+
 // Get attempts to retrieve the most recent revocation for the given cert chain
 // (the  key used in the underlying database is the nodeID of the certificate chain).
-func (r RevocationDB) Get(chain []*x509.Certificate) (*peertls.Revocation, error) {
+func (r RevocationDB) Get(chain []*x509.Certificate) (*extensions.Revocation, error) {
 	nodeID, err := NodeIDFromKey(chain[peertls.CAIndex].PublicKey)
 	if err != nil {
-		return nil, peertls.ErrRevocation.Wrap(err)
+		return nil, extensions.ErrRevocation.Wrap(err)
 	}
 
 	revBytes, err := r.DB.Get(nodeID.Bytes())
 	if err != nil && !storage.ErrKeyNotFound.Has(err) {
-		return nil, peertls.ErrRevocationDB.Wrap(err)
+		return nil, extensions.ErrRevocationDB.Wrap(err)
 	}
 	if revBytes == nil {
 		return nil, nil
 	}
 
-	rev := new(peertls.Revocation)
+	rev := new(extensions.Revocation)
 	if err = rev.Unmarshal(revBytes); err != nil {
-		return rev, peertls.ErrRevocationDB.Wrap(err)
+		return rev, extensions.ErrRevocationDB.Wrap(err)
```

Comment:
to make reviewing easier, it might be good to break name changes out into a separate PR

Reply:
Will be more careful in the future :+1:.

I would've done better git-foo to do it but there was package refactoring happening at the same time to avoid import cycles